x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -54,67 +55,95 @@ export const generateMlInferencePipelineBody = ({
   model,
   pipelineName,
 }: MlInferencePipelineParams): MlInferencePipeline => {
-  // if model returned no input field, insert a placeholder
-  const modelInputField =
-    model.input?.field_names?.length > 0 ? model.input.field_names[0] : 'MODEL_INPUT_FIELD';
-
-  // For now this only works for a single field mapping
-  const sourceField = fieldMappings[0].sourceField;
-  const targetField = fieldMappings[0].targetField;
   const inferenceType = Object.keys(model.inference_config)[0];
-  const remove = getRemoveProcessorForInferenceType(targetField, inferenceType);
-  const set = getSetProcessorForInferenceType(targetField, inferenceType);
-
-  return {
+  const pipelineDefinition: MlInferencePipeline = {
     description: description ?? '',
-    processors: [
-      {
-        remove: {
-          field: getMlInferencePrefixedFieldName(targetField),
-          ignore_missing: true,
-        },
+    processors: [],
+    version: 1,
+  };
+
+  // Add remove and inference processors
+  fieldMappings.forEach(({ sourceField, targetField }) => {
```

Comment:
suggestion: replace `forEach` with a `flatMap`. `forEach` performs poorly, and idiomatic Kibana code is immutable (where reasonable) so maps are preferred. If you want to execute a for loop, using a regular for loop rather than for each is generally a better choice. 

suggestion:
```
const processors = fieldMappings.flatMap(({ sourceField, targetField }) => {
    const remove = getRemoveProcessorForInferenceType(targetField, inferenceType);
    const inference = getInferenceProcessor(
      sourceField,
      targetField,
      inferenceConfig,
      model,
      pipelineName
    );
    return [
    {
      remove: {
        field: getMlInferencePrefixedFieldName(targetField),
        ignore_missing: true,
      },
    },
    ...(remove ? [{ remove }] : []),
    { inference }
    ];
});